include/boost/dll/shared_library.hpp

```diff
@@ -177,7 +177,7 @@ class shared_library
     *
     * \throw Nothing.
     */
-    ~shared_library() BOOST_NOEXCEPT {}
+    virtual ~shared_library() BOOST_NOEXCEPT {}
 
```

Comment:
Some background. I was planning to gather comments on `shared_library` from Boost users, and after that write a proposal for C++ committee to make `shared_library` part of the C++ Standard. C++ committee does not like `virtual`, so it's better to avoid it.

Please revert all the changes to this file. Use `shared_library` as a member of `smart_library`. We could rename the `get_impl` into `get_void` and make it protected. Or you could implment it in `smart_library` like this:
```
shared_library lib_;

void* get_void(const char* name) {
    return boost::addressof(lib_.get<unsigned char>(name));
}
```